src/deploy/functions/runtimes/discovery/index.ts

```diff
@@ -81,7 +81,9 @@ export async function detectFromPort(
   let res: Response;
   const timedOut = new Promise<never>((resolve, reject) => {
     setTimeout(() => {
-      reject(new FirebaseError("User code failed to load. Cannot determine backend specification"));
+      const originalError = 'User code failed to load. Cannot determine backend specification.'
+      const error = `${originalError} Timeout after ${timeout}. Try 'export FUNCTIONS_DISCOVERY_TIMEOUT=${timeout+1}'`
```

Comment:
We now have documentation for this error at https://firebase.google.com/docs/functions/tips#avoid_deployment_timeouts_during_initialization

```suggestion
      const error = `${originalError} Timeout after ${timeout}. See https://firebase.google.com/docs/functions/tips#avoid_deployment_timeouts_during_initialization'`
```